src/hotspot/share/jvmci/jvmciCodeInstaller.cpp

```diff
@@ -760,6 +767,20 @@ JVMCI::CodeInstallResult CodeInstaller::install(JVMCICompiler* compiler,
         nm->maybe_print_nmethod(directive);
         DirectivesStack::release(directive);
       }
+
+      if (nm != nullptr) {
+        if (_nmethod_entry_patch_offset != -1) {
+          BarrierSetNMethod* bs_nm = BarrierSet::barrier_set()->barrier_set_nmethod();
+          if (bs_nm == nullptr) {
```

Comment:
Should never be null. ARM32 was the last to enter the scene but now all platforms have these.